Compilation Errors for mruby 1.4

For mruby 1.4 the gem does not compile anymore:

```
mruby-profiler/src/profiler.c: In function ‘get_class’:
mruby-profiler/src/profiler.c:70:25: error: incompatible types when initializing type ‘struct RProc *’ using type ‘mrb_method_t {aka struct <anonymous>}’
   struct RProc  *proc = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
                         ^~~~~~~~~~~~~~~~~~~~
mruby-profiler/src/profiler.c:83:10: error: incompatible types when assigning to type ‘struct RProc *’ from type ‘mrb_method_t {aka struct <anonymous>}’
     proc = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
          ^
mruby-profiler/src/profiler.c: In function ‘mrb_mruby_profiler_get_irep_info’:
mruby-profiler/src/profiler.c:757:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   int ai = mrb_gc_arena_save(mrb);
   ^~~
mruby-profiler/src/profiler.c: In function ‘mrb_mruby_profiler_gem_init’:
mruby-profiler/src/profiler.c:810:6: error: ‘mrb_state {aka struct mrb_state}’ has no member named ‘code_fetch_hook’
   mrb->code_fetch_hook = prof_code_fetch_hook;
      ^~
mruby-profiler/src/profiler.c: In function ‘get_class’:
mruby-profiler/src/profiler.c:70:25: error: incompatible types when initializing type ‘struct RProc *’ using type ‘mrb_method_t {aka struct <anonymous>}’
   struct RProc  *proc = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
                         ^~~~~~~~~~~~~~~~~~~~
mruby-profiler/src/profiler.c:83:10: error: incompatible types when assigning to type ‘struct RProc *’ from type ‘mrb_method_t {aka struct <anonymous>}’
     proc = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
          ^
mruby-profiler/src/profiler.c: In function ‘mrb_mruby_profiler_get_irep_info’:
mruby-profiler/src/profiler.c:757:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   int ai = mrb_gc_arena_save(mrb);
   ^~~
```

A quick search led me to this commit as the culprit: https://github.com/mruby/mruby/commit/8f2c62407c0659d84126545e19505a851059e750

--- a/src/profiler.c
+++ b/src/profiler.c
@@ -65,9 +65,19 @@ get_class(mrb_state *mrb, struct mrb_irep *irep)
   //Get root class from VM stack
   struct RClass *c    = mrb_class(mrb, mrb->c->stack[0]);
   struct RClass *cc   = c;
+  struct RProc  *proc;
 
   //Get class#method
-  struct RProc  *proc = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
+#if (MRUBY_RELEASE_MAJOR == 1 && MRUBY_RELEASE_MINOR >= 4) || MRUBY_RELEASE_MAJOR > 1
+  #if defined(MRB_METHOD_TABLE_INLINE)
+    mrb_raise(mrb, mrb->eException_class, "profiling not supported when using MRB_METHOD_TABLE_INLINE");
+  #else
+    mrb_method_t meth = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
+    proc = meth.proc;
+  #endif
+#else
+  proc = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
+#endif
   const  mrb_sym m    = mrb_intern_cstr(mrb, "to_s");
 
   //If method couldn't be retrieved or it's invalid, just use the original class
@@ -80,7 +90,16 @@ get_class(mrb_state *mrb, struct mrb_irep *irep)
   //While the method definition doesn't match, try superclasses
   while(proc->body.irep != irep) {
     cc = cc->super;
+#if (MRUBY_RELEASE_MAJOR == 1 && MRUBY_RELEASE_MINOR >= 4) || MRUBY_RELEASE_MAJOR > 1
+    #if defined(MRB_METHOD_TABLE_INLINE)
+      mrb_raise(mrb, mrb->eException_class, "profiling not supported when using MRB_METHOD_TABLE_INLINE");
+    #else
+      meth = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
+      proc = meth.proc;
+    #endif
+#else
     proc = mrb_method_search_vm(mrb, &cc, mrb->c->ci->mid);
+#endif
     if(!cc || !proc || !mrb_respond_to(mrb, mrb_obj_value(cc), m)) {
       return TO_S(c);
     }

